pkg/controller/utils/utils.go

```diff
@@ -114,6 +115,34 @@ func AddServiceWatch(c controller.Controller, name, namespace string) error {
 	})
 }
 
+func addLicenseWatch(c controller.Controller) error {
+	lic := &v3.LicenseKey{
+		TypeMeta: metav1.TypeMeta{Kind: "LicenseKey"},
+	}
+	return c.Watch(&source.Kind{Type: lic}, &handler.EnqueueRequestForObject{})
+}
+
+// WaitToAddLicenseKeyWatch will check if the API server is available and if so, it will add a watch for LicenseKey
+// The completion of this operation will be signaled on a ready channel
+func WaitToAddLicenseKeyWatch(controller controller.Controller, client client.Client, log logr.Logger, ready chan bool) {
+	ticker := time.NewTicker(10 * time.Minute)
```

Comment:
This is a long time to wait - if we miss the initial watch on this couldn't it delay rollout of some components an additional 10 minutes?

Is there a way we could be more responsive without doing tons of GET requests? Maybe by watching the apiserver CR status?